validator/sawtooth_validator/journal/publisher.py

```diff
@@ -370,7 +370,8 @@ def _finalize_block(self, block):
                 else:
                     block.add_batch(batch)
                     self._committed_txn_cache.add_batch(batch)
-                state_hash = result.state_hash
+                if result.state_hash is not None:
+                    state_hash = result.state_hash
```

Comment:
Is this check necessary? Does this handle the case where a transaction does not update the Merkle Trie?